source/draw/gpu/opengl/backend/gles3/Gles3ShaderProgram.ooc

```diff
@@ -86,6 +87,31 @@ Gles3ShaderProgram: class extends GLShaderProgram {
 		glUniform4f(glGetUniformLocation(this _backend, name), value x, value y, value z, value w)
 		version(debugGL) { validateEnd("ShaderProgram setUniform~FloatPoint4D") }
 	}
+	setUniform: override func ~ColorBgr (name: String, value: ColorBgr) {
+		version(debugGL) { validateStart("ShaderProgram setUniform~ColorBgr") }
+		glUniform3f(glGetUniformLocation(this _backend, name), (value blue as Float) / 255.0f, (value green as Float) / 255.0f, (value red as Float) / 255.0f)
```

Comment:
I'm not talking about leaking it, I'm talking about moving the conversions to a higher level (but still not in the public API). We already have a OpenGL-specific type check for every shader variable upload value, any conversions to OpenGL could be made there. Then we keep the low level ShaderProgram API clean and primitive (as it should be).